File: src/lib/pjecalc/core/dominio/termo/comportamento/comportamento-media-pelo-valor-corrigido.ts
/**
 * Porte 1:1 de ComportamentoMediaPeloValorCorrigido.java (21 linhas).
 *
 * Wrapper: chama ComportamentoMediaPeloValor com flag `corrigir=true`.
 *
 * Ref: pjecalc-fonte/.../dominio/termo/comportamento/ComportamentoMediaPeloValorCorrigido.java
 */
import type Decimal from 'decimal.js';
import { ComportamentoMediaPeloValor } from './comportamento-media-pelo-valor';
import type { ItemBaseVerba } from '../item-base-verba';
import type { ParametroDoTermo } from '../parametro-do-termo';

export class ComportamentoMediaPeloValorCorrigido extends ComportamentoMediaPeloValor {
  resolverValor(item: ItemBaseVerba, parametro: ParametroDoTermo): Decimal {
    return this.resolverValorInterno(item, parametro, true);
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/termo/comportamento/ComportamentoMediaPeloValorCorrigido.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.termo.comportamento;

import br.jus.trt8.pjecalc.negocio.dominio.termo.ItemBaseVerba;
import br.jus.trt8.pjecalc.negocio.dominio.termo.ParametroDoTermo;
import br.jus.trt8.pjecalc.negocio.dominio.termo.comportamento.ComportamentoMediaPeloValor;
import java.math.BigDecimal;

public class ComportamentoMediaPeloValorCorrigido
extends ComportamentoMediaPeloValor {
    private static final long serialVersionUID = 3814765900817121927L;
    private static final boolean CORRIGIR = Boolean.TRUE;

    @Override
    public BigDecimal resolverValor(ItemBaseVerba item, ParametroDoTermo parametro) {
        return super.resolverValor(item, parametro, CORRIGIR);
    }
}

